alerting/src/main/kotlin/org/opensearch/alerting/DocumentLevelMonitorRunner.kt

```diff
@@ -598,17 +578,25 @@ object DocumentLevelMonitorRunner : MonitorRunner() {
         return allShards.filter { it.primary() }.size
     }
 
-    private suspend fun getMatchingDocs(
+    /** 1. Fetch data per shard for given index. (only 10000 docs are fetched. needs to be converted to scroll if performant enough)
+     *  2. Transform documents to submit to percolate query
+     *  3. perform percolate queries
+     *  4. update docToQueries Map with all hits from percolate queries
+     *  */
+    private suspend fun fetchDataAndExecutePercolateQueriesPerShard(
         monitor: Monitor,
         monitorCtx: MonitorRunnerExecutionContext,
         docExecutionCtx: DocumentExecutionContext,
-        index: String,
-        concreteIndex: String,
+        indexName: String,
+        concreteIndexName: String,
         conflictingFields: List<String>,
-        docIds: List<String>? = null
-    ): List<Pair<String, BytesReference>> {
+        docIds: List<String>? = null,
+        monitorMetadata: MonitorMetadata,
+        inputRunResults: MutableMap<String, MutableSet<String>>,
+        docsToQueries: MutableMap<String, MutableList<String>>,
+    ) {
         val count: Int = docExecutionCtx.updatedLastRunContext["shards_count"] as Int
-        val matchingDocs = mutableListOf<Pair<String, BytesReference>>()
+        val transformedDocs = mutableListOf<Pair<String, BytesReference>>()
```

Comment:
Should this go in the shard iterating for-loop? If I'm reading this right, we are adding the docs from each shard to the list on each loop iteration without resetting the list in between. Then executing the percolator query against the docs from the current shard + all previous shards on each loop

Reply:
thanks for catching this. will fix it